extensions/amp-share-tracking/0.1/amp-share-tracking.js

```diff
@@ -128,8 +131,34 @@ export class AmpShareTracking extends AMP.BaseElement {
    * @return {!Promise<string>}
    * @private
    */
-  getRandomBase64_() {
-    return Promise.resolve(base64UrlEncodeFromBytes(getRandomBytesArray(6)));
+  getShareTrackingRandomBase64_() {
+    return Promise.resolve(base64UrlEncodeFromBytes(
+        this.getShareTrackingRandomBytes_()));
+  }
+
+  /**
+   * Get a random bytes array that has 48 bits (6 bytes).
+   * Use win.crypto.getRandomValues if it is available.
+   * Otherwise, use Math.random as fallback.
+   * @return {!Promise<string>}
+   * @private
+   */
+  getShareTrackingRandomBytes_() {
+    // Use win.crypto.getRandomValues to get 48 bits of random value
+    let bytes = getCryptoRandomBytesArray(this.win,
+        SHARE_TRACKING_NUMBER_OF_BYTES); // 48 bit
+
+    // Support for legacy browsers
+    if (!bytes) {
+      bytes = new Uint8Array(SHARE_TRACKING_NUMBER_OF_BYTES);
+      let random = Math.random();
+      for (let i = 0; i < SHARE_TRACKING_NUMBER_OF_BYTES; i++) {
+        random <<= 8;
```

Comment:
This is wrong. I want to shift left 8 bits, but it's actually not doing it. I could use `random *= 256;` instead. I'll change it.